api/src/main/java/com/velocitypowered/api/proxy/InboundConnection.java

```diff
@@ -43,4 +44,11 @@ public interface InboundConnection {
    * @return the protocol version the connection uses
    */
   ProtocolVersion getProtocolVersion();
+
+  /**
+   * Returns the current protocol state of this connection.
+   *
+   * @return the protocol state of the connection
+   */
+  ProtocolState protocolState();
```

Comment:
I don't like to mix record style getters and the other getters in the same class. Either change the style of all getters and deprecate the old ones or change this to getProtocolState();

Reply:
While I would like to change all the methods to Adventure/records style, I can't do it just before the release, I will leave it as a regular getter, until the next major release